drivers/flash/flash_stm32_qspi.c

```diff
@@ -886,6 +892,121 @@ static void flash_stm32_qspi_pages_layout(const struct device *dev,
 }
 #endif
 
+#if defined(CONFIG_FLASH_EX_OP_ENABLED)
+#if defined(CONFIG_FLASH_STM32_QSPI_GENERIC_READ)
+static int flash_stm32_qspi_generic_read(const struct device *dev, QSPI_CommandTypeDef *cmd,
+					 void *out)
+{
+	int ret;
+
+	void *buffer = out;
+
+#ifdef CONFIG_USERSPACE
+	QSPI_CommandTypeDef cmd_copy;
+	uint8_t *out_copy;
+
+	bool syscall_trap = z_syscall_trap();
+
+	if (syscall_trap) {
+		K_OOPS(k_usermode_from_copy(&cmd_copy, cmd, sizeof(cmd_copy)));
+		cmd = &cmd_copy;
+
+		if (cmd->NbData <= CONFIG_FLASH_STM32_QSPI_GENERIC_STACK_BUFFER_MAX_SIZE) {
+			out_copy = alloca(CONFIG_FLASH_STM32_QSPI_GENERIC_STACK_BUFFER_MAX_SIZE);
+		} else {
+			out_copy = k_malloc(CONFIG_FLASH_STM32_QSPI_GENERIC_STACK_BUFFER_MAX_SIZE);
+			if (out_copy == NULL) {
+				return -ENOMEM;
+			}
+		}
+
+		buffer = out_copy;
+	}
+#endif
+	qspi_lock_thread(dev);
+
+	ret = qspi_read_access(dev, cmd, buffer, cmd->NbData);
+
+	qspi_unlock_thread(dev);
+
+#ifdef CONFIG_USERSPACE
+	if (syscall_trap) {
+		K_OOPS(k_usermode_to_copy(out, buffer, cmd->NbData));
+
+		if (cmd->NbData > CONFIG_FLASH_STM32_QSPI_GENERIC_STACK_BUFFER_MAX_SIZE) {
+			k_free(out_copy);
+		}
+	}
+#endif
+	return ret;
+}
+#endif /* CONFIG_FLASH_STM32_QSPI_GENERIC_READ */
+
+#if defined(CONFIG_FLASH_STM32_QSPI_GENERIC_WRITE)
+static int flash_stm32_qspi_generic_write(const struct device *dev, QSPI_CommandTypeDef *cmd,
+					  void *in)
+{
+	int ret;
+
+#ifdef CONFIG_USERSPACE
+	QSPI_CommandTypeDef cmd_copy;
+	uint8_t *in_copy;
+
+	bool syscall_trap = z_syscall_trap();
+
+	if (syscall_trap) {
+		K_OOPS(k_usermode_from_copy(&cmd_copy, cmd, sizeof(cmd_copy)));
+		cmd = &cmd_copy;
+
+		if (cmd->NbData <= CONFIG_FLASH_STM32_QSPI_GENERIC_STACK_BUFFER_MAX_SIZE) {
+			in_copy = alloca(CONFIG_FLASH_STM32_QSPI_GENERIC_STACK_BUFFER_MAX_SIZE);
+		} else {
+			in_copy = k_malloc(CONFIG_FLASH_STM32_QSPI_GENERIC_STACK_BUFFER_MAX_SIZE);
```

Comment:
Isn't the intention here to allocate `cmd->NbData` bytes instead of `CONFIG_FLASH_STM32_QSPI_GENERIC_STACK_BUF` (_`else` branch_)?